base-system/usrroot/usr/sbin/ipman

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+#	ipman.sh
+#	This script sets the IP address of the huronOS instance based on the
+#   configuration file (sync-server.conf) and if no interface sticks with
+#   the defined IP it leaves the available interfaces with dhcp.
+#
+#	Copyright (C) 2022, huronOS Project:
+#		<http://huronos.org>
+#
+#	Licensed under the GNU GPL Version 2
+#		<http://www.gnu.org/licenses/gpl-2.0.html>
+#
+#	Authors:
+#		Daniel Cerna <dcerna@huronos.org>
+
+. /usr/lib/hos/enviroment.sh
+
+# Retrieves the required fields
+IP_ADDRESS=$(grep "^IP_ADDRESS=.*$" $CURRENT_SYNC_SERVER_CONFIG_FILE | cut -d= -f2)
+IP_MASK=$(grep "^IP_MASK=.*$" $CURRENT_SYNC_SERVER_CONFIG_FILE | cut -d= -f2)
+IP_GATEWAY=$(grep "^IP_GATEWAY=.*$" $CURRENT_SYNC_SERVER_CONFIG_FILE | cut -d= -f2)
+IP_SERVER=$(grep "^DIRECTIVES_SERVER_IP=.*$" $CURRENT_SYNC_SERVER_CONFIG_FILE | cut -d= -f2)
+DIRECTIVES_FILE_URL=$(grep "^DIRECTIVES_FILE_URL=.*$" $CURRENT_SYNC_SERVER_CONFIG_FILE | cut -d= -f2)
+
+# Convert subnet mask decimal form to the subnet mask octal form
+ONES=$(( 2**32 - 2**(32 - IP_MASK) ))
+SUBNET_MASK=$(printf "%d.%d.%d.%d\n" $((ONES >> 24 & 255)) $((ONES >> 16 & 255)) $((ONES >> 8 & 255)) $((ONES & 255)))
+
+
+# Test if any of the required fields is empty
+if [ -z "$IP_ADDRESS" ]; then
+    echo "No IP defined, exiting"
+    exit 0
+fi
+if [ -z "$IP_MASK" ]; then
+    echo "No ip mask defined, exiting"
+    exit 0
+fi
+if [ -z "$IP_GATEWAY" ]; then
+    echo "No gateway defined, exiting"
+    exit 0
+fi
+
+# Get a list of all available interfaces
+INTERFACES=$(connmanctl services | awk '{print $3}')
```

Comment:
What happens with Wifi interfaces? I don't think the config will ever work because we don't have any network passwords config.
Maybe we can shrink the scope to only ethernet¿, we know all ethernet interfaces can only start the `e`

Reply:
Done by grepping to lines starting with e